XML to JSON mapping for sections 0, 1, and 2

Samarth-Agarwal1 opened the pull request:
@SoniaZaldana I have added a file that specifies the XML to JSON mapping for sections 0, 1 and 2. 

Three important things I want to point out:
1. The `id` field of each question in this JSON form is the `name` field of the corresponding question in the XML form (https://github.com/SoniaZaldana/covid19-reporting/blob/jsonConversion/src/form.json). It is NOT the `ID` field in the XML form. The reason I did this is because the values for `ID` in the XML form seem to be floats and it seems to make more sense to use the values for `name` in the XML form since they are shorter and alphanumeric.
2. The `question` field of each question in this JSON form is the `title` field of the corresponding question in the XML form, NOT the PDF (https://github.com/IHE-SDC-WG/SDC-Sample-XML-Files/blob/master/COVID%2019/2019-covid-crf-v6.pdf). The questions in the PDF and the value in the `title` field of the XML form aren't always equivalent, so I decided to go with the value in the `title` field when choosing what to write in the `question` field for the JSON.
3. There are two places in the JSON that deviate from the structure as we discussed in the meeting and I have commented in the JSON where the two places are and why I did it that way.

One thing that I did that I'm unsure of:
In the JSON mapping, under `preliminary_info`, I created this object:
```
{
                "id": "Q_35849",
                "question": "Why Tested for COVID-19",
                "response": <list of string>,
                "followup_response": [
                    {
                        "id": "LI_35963",
                        "question": "None of the above (explain)",
                        "response": <string>
                    }
                ]
            }
```

I'm not sure if there should be a `followup_response` here. The reason I added it in is because if you look in the PDF (https://github.com/IHE-SDC-WG/SDC-Sample-XML-Files/blob/master/COVID%2019/2019-covid-crf-v6.pdf), at the top of the page where it says "Why tested for COVID-19:", one of the options is "If none of the above, please explain". I'm not sure if the response to this option should go as a followup response or be a part of the intial `response`. What do you think?

Keeping the above things I mentioned in mind, you should be good to start on mapping your sections! Let me know if you have any questions/need clarification!

SoniaZaldana commented on JSON_mapping.md:
```diff
@@ -0,0 +1,190 @@
+                                                    // XML form. So this object will represent the units of the age (days, months or years).
+                        "id": "Q_35972_Units",
+                        "question": "Age Units",
+                        "response": <string>        // value for "response" should be one of the following: {"days", "months", "years"}
```
You could probably add this as the body of "response": 
```
{
  "days": <int>, 
  "months": <int>, 
  "years": <int>
}
```

SoniaZaldana replied:
You could probably add this as the body of "response": 
```
{
  "days": <int>, 
  "months": <int>, 
  "years": <int>
}
```

SoniaZaldana commented on JSON_mapping.md:
```diff
@@ -0,0 +1,190 @@
+            {
+                "id": "Q_38610",
+                "question": "Underlying Conditions and Comorbidity",
+                "response": <string>,
```
Wouldn't the response here be a list of strings?

Samarth-Agarwal1 replied:
So, this question corresponds to this question in the PDF:
![image](https://user-images.githubusercontent.com/24865141/139591201-c7aff432-a1f6-4fe0-87a8-9d79e30fe368.png)
So, the "response" will be either "No", "Yes" or "Unknown".

SoniaZaldana commented on JSON_mapping.md:
```diff
@@ -0,0 +1,190 @@
+                "question": "Underlying Conditions and Comorbidity",
+                "response": <string>,
+                "followup_response": [
+                    {                               // Note that this object doesn't have a "question" field because the corresponding question in the XML form doesn't have
```
I'm not sure this followup response object is necessary if the response object above just has a list of all the conditions.

Samarth-Agarwal1 replied:
And this followup response corresponds to this question in the PDF:
![image](https://user-images.githubusercontent.com/24865141/139591268-b30df188-03a7-4ce4-ac88-466b537a81c7.png)

SoniaZaldana reviewed (changes requested):
Looks great @Samarth-Agarwal1! Just left some tiny comments. I have already started work on my own sections but I will wait to open a PR until we get this PR worked out so I can pull any changes. 

I was also wondering if we should create a .json file with the full JSON once we complete this? So that we can use that file to load it into our program and populate it at run time.

Samarth-Agarwal1 commented:
@SoniaZaldana Thanks for reviewing this! I've changed the response for the age question to be an object corresponding to the age units as you suggested. For the other two points you made, I replied to them. 

One more question I have is:
In the JSON mapping, under `preliminary_info`, I created this object:
```
{
                "id": "Q_35849",
                "question": "Why Tested for COVID-19",
                "response": <list of string>,
                "followup_response": [
                    {
                        "id": "LI_35963",
                        "question": "None of the above (explain)",
                        "response": <string>
                    }
                ]
            }
```

I'm not sure if there should be a followup_response here. The reason I added it in is because if you look in the PDF (https://github.com/IHE-SDC-WG/SDC-Sample-XML-Files/blob/master/COVID%2019/2019-covid-crf-v6.pdf), at the top of the page where it says "Why tested for COVID-19:", one of the options is "If none of the above, please explain". I'm not sure if the response to this option should go as a followup response or be a part of the intial response. What do you think?
